src/union.ts

```diff
@@ -330,18 +331,121 @@ export class Union {
      * @param fs the filesystem interface to be added to the queue of FS's
      * @returns this instance of a unionFS
      */
-    use(fs: IFS): this {
-        this.fss.push(fs);
+    use(fs: IFS, options: VolOptions = {}): this {
+        this.fss.push([fs, options, this.createMethods(fs, options)]);
         return this;
     }
 
+    /**
+     * At the time of the [[use]] call, we create our sync, async and promise methods
+     * for performance reasons
+     *
+     * @param fs
+     * @param options
+     */
+    private createMethods(fs: IFS, options: VolOptions): FSMethodStack {
+        const noop = undefined
+        const createFunc = (method: string) => {
+            if (!fs[method]) return (...args: any[]) => { throw new Error(`Method not supported: "${method}" with args "${args}"`); };
+            return (...args: any[]) => fs[method as string].apply(fs, args);
+        }
+        switch (true) {
```

Comment:
The cases here are all very similar, so you could simplify things by just doing the checks in the reduces.

i.e

```
fsSyncMethodsReadonly.reduce((acc, method) => {acc[method] = options.writeonly ? noop : createFunc(method); return acc;}, {}),
```

This also means you can use both `readonly` & `writeonly` together (which I think should be the default, after some tweaking - see my other comments for more)

Reply:
ah yes! Much simpler